src/twisted/names/common.py

```diff
@@ -210,20 +240,23 @@ def extractRecord(resolver, name, answers, level=10):
                 resolver, r.payload.name, answers, level - 1)
             if not result:
                 return resolver.getHostByName(
-                    str(r.payload.name), effort=level - 1)
+                    r.payload.name.name, effort=level - 1)
             return result
     # No answers, but maybe there's a hint at who we should be asking about
     # this
     for r in answers:
-        if r.type == dns.NS:
-            from twisted.names import client
-            r = client.Resolver(servers=[(str(r.payload.name), dns.PORT)])
-            return r.lookupAddress(str(name)
-                ).addCallback(
-                    lambda records: extractRecord(
-                        r, name,
-                        records[_ANS] + records[_AUTH] + records[_ADD],
-                        level - 1))
+        if r.type != dns.NS:
+            continue
+        from twisted.names import client
```

Comment:
I know the code isn't new, but I should probably point out that doing the local import *inside* a hot loop is probably not a great idea…